approval_polls/views.py

```diff
@@ -79,7 +79,7 @@ def add(request):
     if not len(choices):
       return render(request, 'approval_polls/error.html', {'error_message': 'At least one choice is required'})
 
-    p = Poll(question=question, pub_date=timezone.now())
+    p = Poll(question=question, pub_date=timezone.now(), username=request.POST['username'].strip())
```

Comment:
Hi @ashimaathri,

I've done some research. The solution I have found is to use request.user instead of  request.session, and for the model I use to_filed to specify which field the foreign key to point to (since username is not the primary key).
Please let me know if I am not on the right track.
Thanks.